content/addons.md

```diff
@@ -138,20 +146,216 @@ There are several options to see the addon in action. We could use `npm link` or
+Check to make sure that your `package.json` is valid, looking for missing commas or trailing commas. "Template precompiler" errors mean that we forgot Step One. `404 not found` means we forgot to `yarn` or `npm install`. Other errors are likely due to file naming problems. For example, trying to rename an addon or component after it has been created is prone to mistakes. And of course, we need to make sure we saved all the files that we changed along the way. (The author of this guide made every single mistake in this list while writing it.)
+
+### Making a UI component available in block form
+
+In an Ember app, components can be used in ["simple" or "block" form](https://guides.emberjs.com/release/components/wrapping-content-in-a-component/). Addon templates have the same capabilities. Simple form is useful when the app developer should providing data objects or configuration values to the addon, while the block form is most useful when the developer should also be able to pass in some of their own templating, content, and interactivity.
+
+In an Ember app, a block style component uses the `{{yield}}` helper as a placeholder for where the passed-in content will go. It is the same in an Ember addon. 
+
+Let's change our button addon we made earlier so that developers can pass in their own handlebars content by using the `{{yield}}` helper:
+
+```hbs
+<!-- addon/templates/components/<addon-name>.hbs -->
+
+<button>{{yield}}</button>
+```
+
+Now, an app can use the addon with their own content inside:
+
+```hbs
+<!-- This is a handlebars file in the app using the addon -->
+
+{{#addon-name}}
+  Register <img href="./images/some-cute-icon.png" alt="">
+{{/addon-name}}
+```
+
+Whatever goes inside the block form addon will show up where the `{{yield}}` was. This is the markdown that renders in the app:
+
+```html
+<!-- markdown rendered by running the app -->
+
+<button>
+  Register <img href="./images/some-cute-icon.png" alt="">
+</button>
+```
+
+### Styling a UI component addon
+
+Addon developers have many options for handling styles within their addons. For example, we could stick to plain old CSS, or use a preprocessor like Less or Sass. Most seasoned addon authors prefer SASS/SCSS. We could automatically style the UI elements when they are used in an app, or we could let the developer who installed the addon choose which stylesheets to include. Here are a few different approaches. Luckily, the Ember CLI handles most of the work for us and we don't have to worry about the inner workings of asset compilation.
```

Comment:
"Sass" is written with only the first letter capitalized first, then with all letters capitalized the second time. I could see a reason being that the second refers to the file extension (since "Scss" is never written that way). But I think it might be clearer to actually write out ".sass/.scss" as lower-cases extensions, or else just say that "most addon authors prefer Sass"